components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/AlertSubscriptionsApiServiceImpl.java

```diff
@@ -138,6 +163,88 @@ public Response getSubscribedAlertTypes(MessageContext messageContext) {
         return Response.status(Response.Status.OK).build();
     }
 
+    /**
+     * Retrieve a list of bot detection alert subscriptions
+     *
+     * @param messageContext
+     * @return list of bot detection alert subscriptions
+     * @throws APIManagementException
+     */
+    @Override
+    public Response getBotDetectionAlertSubscriptions(MessageContext messageContext) throws APIManagementException {
+
+        List<BotDetectionData> botDetectionDataList = new APIAdminImpl().retrieveSavedBotDataEmailList();
+        BotDetectionAlertSubscriptionListDTO listDTO =
+                BotDetectionMappingUtil.fromAlertSubscriptionListToListDTO(botDetectionDataList);
+        return Response.ok().entity(listDTO).build();
+    }
+
+    /**
+     * Subscribe for bot detection alerts
+     *
+     * @param body           email to be registered for the subscription
+     * @param messageContext
+     * @return alert subscription DTO containin the uuid of the subscription and the registered email
+     * @throws APIManagementException
+     */
+    @Override
+    public Response subscribeForBotDetectionAlerts(BotDetectionAlertSubscriptionDTO body, MessageContext messageContext)
+            throws APIManagementException {
+
+        String email = body.getEmail();
+        if (StringUtils.isBlank(email)) {
+            String propertyName = AlertMgtConstants.BOT_DETECTION_EMAIL_FIELD;
+            throw new APIManagementException(propertyName + " property value of payload cannot be blank",
+                    ExceptionCodes.from(ExceptionCodes.BLANK_PROPERTY_VALUE, propertyName));
+        }
+
+        try {
+            APIAdminImpl apiAdmin = new APIAdminImpl();
+            BotDetectionData alertSubscription =
+                    apiAdmin.getBotDetectionAlertSubscription(AlertMgtConstants.BOT_DETECTION_EMAIL_FIELD, email);
+            if (alertSubscription != null) {
+                RestApiUtil.handleResourceAlreadyExistsError(
+                        "Email: " + email + " has already been subscribed for bot detection alerts", log);
+            }
+            apiAdmin.addBotDataEmailConfiguration(email);
+            BotDetectionData newAlertSubscription =
+                    apiAdmin.getBotDetectionAlertSubscription(AlertMgtConstants.BOT_DETECTION_EMAIL_FIELD, email);
+            BotDetectionAlertSubscriptionDTO alertSubscriptionDTO =
+                    BotDetectionMappingUtil.fromAlertSubscriptionToDTO(newAlertSubscription);
+            return Response.ok(alertSubscriptionDTO).build();
+        } catch (APIManagementException | SQLException e) {
```

Comment:
SQLException should not be coming upto REST API level. Shall we check why it is coming here?

Reply:
Also, you don't need to catch APIManagementException, just let it pass. It will give a 500 anyway.
https://github.com/wso2/carbon-apimgt/wiki/Exception-handling-in-REST-APIs

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/8791/commits/110c56498df5d7c1218d97b1dfda9e179aa6fee4